lib/remote-connector.js

```diff
@@ -91,14 +129,56 @@ function createProxyMethod(Model, remotes, remoteMethod) {
       callback = args.pop();
     }
 
+    // Take care not to overwrite method args
+    var numberOfDefaultArgs = remoteMethod.accepts.length - headerArgs.length;
+    headerArgs.forEach(function(headerArg, idx) {
+      args[idx + numberOfDefaultArgs] = headerArg;
+    });
+
+    if (auth && auth.accessTokenId) {
+      // Find correct index for auth header
+      remoteMethod.accepts.forEach(function(accept, idx) {
+        if (accept.name === 'authorization') {
+          args[idx] = auth.accessTokenId;
+        }
+      });
+    }
+
+    // Make sure gaps in array are numbered.
+    for (var i = 0, argsLength = args.length; i < argsLength; i++) {
+      args[i] = args[i] || void 0;
+    }
+
     if (remoteMethod.isStatic) {
       return remotes.invoke(remoteMethod.stringName, args, callback);
     }
 
     var ctorArgs = [this.id];
     return remotes.invoke(remoteMethod.stringName, ctorArgs, args, callback);
-  }
+  };
 }
 
-function noop() {
+function interceptUserRemotes(remotes, auth) {
+  remotes.after('User.logout', function(ctx, next) {
+    auth.clearUser();
+    auth.clearStorage();
+    next();
+  });
+
+  remotes.before('User.login', function(ctx, next) {
+    auth.rememberMe = ctx.req.body.rememberMe || false;
```

Comment:
Perhaps we ca add `rememberMe` to `User.login` signature?

```js
User.login = function(credentials, rememberMe, include, fn) { }
```

That way the connector should be able to pull the value from the context in a transport-agnostic way.

---

On the other hand, my understanding is that `rememberMe` is set via the `credentials` argument at the moment, right?

```js
User.login({ email, password, rememberMe }, cb);
```

In that case you should be able to read the value of the `credentials` argument directly from the context in a transport-agnostic way, and extract `rememberMe` from there. IMO it's ok to keep this extra property in the body of the request sent to the server, strong-remoting and/or `User.login` will simple ignore this unknown argument.